web/src/condense.ts

```diff
@@ -169,9 +170,21 @@ export function condense_and_collapse(elems: JQuery): void {
         return;
     }
 
-    const height_cutoff = message_viewport.max_message_height();
-    const rows_to_resize = [];
+    // For unread messages, we allow them to expand to most of a
+    // desktop monitor's height, with stricter limits for mobile web
+    // devices, especially in landscape mode.
 
+    const estimated_header_plus_footer_height = 160;
+    const height_cutoff_unread = Math.min(
+        30 * user_settings.web_font_size_px,
+        message_viewport.height() - estimated_header_plus_footer_height,
+    );
```

Comment:
The last proposal [in the chat thread](https://chat.zulip.org/#narrow/channel/101-design/topic/when.20to.20condense.20messages/near/1994418) is for this to be the `max` of these two arguments, rather than `min`.

Reply:
Oops, updated to use `max`.